codecs/rotate/rotate.rs

```diff
@@ -69,13 +83,8 @@ fn rotate(input_width: isize, input_height: isize, rotate: isize) {
   for d2 in 0..d2_limit {
     for d1 in 0..d1_limit {
       let in_idx = (d1_start + d1 * d1_advance) * d1_multiplier + (d2_start + d2 * d2_advance) * d2_multiplier;
-      out_b[i as usize] = in_b[in_idx as usize];
+      *unwrap_abort(out_b.get_mut(i as usize)) = *unwrap_abort(in_b.get(in_idx as usize));
```

Comment:
Won't this not compile in debug mode? I think you meant to name the `unchecked_unwrap` function `unwrap_abort` (and also make it non-unsafe).

This also looks pretty ugly. May I recommend an extension trait:
```rust
trait UnwrapHard {
  type Output;
  fn unwrap_hard(self) -> Self::Output;
}

impl<T> UnwrapHard for Option<T> {
    type Output = T;
    #[cfg(not(debug_assertions))]
    #[inline]
    fn unwrap_hard(self) -> T {
        use std::process;
        match self {
            Some(t) => t,
            None => process::abort(),
        }
    }
    
    #[cfg(debug_assertions)]
    fn unwrap_hard(self) -> T {
        self.unwrap()
    }
}
```
To see it in action, see [the playground](https://play.rust-lang.org/?version=stable&mode=release&edition=2018&gist=198d299a0ebeb95c488a1971aed57233)

Reply:
You are correct. The unsafe/misnaming was an oversight copy/pasting from @fitzgen lol. I’m working on a new PR anyways and we shipped the release build, so we are good for now 😅

Reply:
Also good shout on the trait. I’m gonna use that in the next PR. Thank you very much!